### ISSUE TITLE
“Who are you and what do you want?” not parsed correctly

### ISSUE THREAD

**Issue by ProgVal** [ProjetPP/PPP-QuestionParsing-Grammatical#73]
http://askplatyp.us/?lang=en&q=Who+are+you+and+what+do+you+want%3F

**Comment by Ezibenroc** [comment]
I don't think we should focus on such questions (i.e. "What is the size of an ant and who is the president of the United States?"). These are really stupid...

Moreover, if we wanted to parse it correctly, we would have to output an union and not an intersection, which would be a bad choice for a lot of more interesting questions (we already discussed it).

**Comment by ProgVal** [comment]
When there are two question words, separated by an “and”, I think you could guess it's a union

**Comment by Ezibenroc** [comment]
Mentionned in https://github.com/ProjetPP/Documentation/pull/48#issuecomment-71982532

I find the questions such as "What are the capital of France and the recipe for cookies?" irrelevant.

Moreover, if for some obscure reason we still want to answer to these questions, this cannot be done by this module, due to the choice we made : "and" are intersections and "or" are unions.
This choice is more relevant in a majority of use cases.
The previous question is grammaticaly equivalent to "Who are the children of Nicolas Sarkozy and Carla Bruni", so there is no hope to be abble to distinguish the cases where "and" should be an union or an intersection.

@yhamoudi what do you think?

**Comment by Ezibenroc** [comment]
Better example:
"Who are the children of Nicolas Sarkozy and Carla Bruni?" ← intersection
vs
"What are the birth dates of Nicolas Sarkozy and Carla Bruni?" ← union

**Comment by yhamoudi** [comment]
It could be intersting to handle multiple questions. However, I'm not sure that using the "union" operator of the datamodel is the right way to do (at the end, there is no distinction between the answers, ex: "What are the capitals of France and Germany?" -> [Paris,Berlin] -> we do not know which one is the capital of France).

I think there could be a preprocessing operation that splits the questions in many entities ("What are the capitals of France and Germany?" > "What is the capital of France?" + "What is the capital of Germany") and then each question is processed individualy.

We could handle (into the question parsing) “Who are you and what do you want?”, but i dislike adding heuristics for such strange cases.

**Comment by Ezibenroc** [comment]
Good remark.
I think that this would imply a modification of the datamodel, if we want to keep a link between each question and the associated answers, within a single query.

**Comment by yhamoudi** [comment]
Not sure (perhaps it's the task of the core to deal with multiples questions and the task of the web ui to display properly the answers to each question). 

I think we can always split the question without losing any information. For instance if the question is "Who is the prime minister of United Kingdom and where does he live?" it could become : "Who is the prime minister of United Kingdom?" + "Where does the prime minister of United Kingdom live?"

See `coreference resolution` and... the [stanford parser](http://nlp.stanford.edu/software/dcoref.shtml)
